lib/bap_llvm/bap_llvm_ogre_coff.mli

```diff
@@ -0,0 +1,33 @@
+
+module Scheme : sig
+  (** coff section in file *)
+  val section_header :
+    (string * int64 * int64, (string -> int64 -> int64 -> 'a) -> 'a)
+      Ogre.attribute
+
+  (** coff section in memory *)
+  val virtual_section_header :
+    (string * int64 * int64, (string -> int64 -> int64 -> 'a) -> 'a)
+      Ogre.attribute
+
+  (** coff section access flags *)
+  val section_flags :
+    (string * (bool * bool * bool),
+     (string -> bool -> bool -> bool -> 'a) -> 'a)
+      Ogre.attribute
+
+  (** coff section that contains code *)
+  val code_content : (string, (string -> 'a) -> 'a) Ogre.attribute
+
+  (** coff symbol *)
+  val symbol :
+    (string * int64 * int64, (string -> int64 -> int64 -> 'a) -> 'a)
+      Ogre.attribute
+
+  (** coff symbol is function *)
+  val function_ : (string * int64, (string -> int64 -> 'a) -> 'a) Ogre.attribute
+end
+
+module Make(Fact : Ogre.S) : sig
```

Comment:
So, this is a naming/interface issue. What is `image` is totally non-obvious, as the interface is not clearly specified, it is kind of internal and is known only to you. On the other hand, we have `segment`, `section` and `symbols` and `mapped` propositions, that are documented and well understood. The purpose of the `Make(Fact)` functor is to derive these well-known properties from the properties that are specific to a concrete file format. The deep understanding of the `Make(Fact).{segments,symbols,sections}` functions is that they add rules to our logic database. Thus, my suggestion is to use the following interface:

```ocaml
module Rules : sig
  module type S = sig
    type 'a m
    val segments : unit m
    val sections : unit m
    val symbols : unit m
  end 
  module Make(Fact : Ogre.S) : S with type 'a m := 'a Fact.t
  include S with type 'a m = 'a
end
```

(the `Rules.S` module type can be actually reused by different backends)

And the interface is supposed to be used as follows:

```ocaml
module Derive = Rules.Make(Fact)
let image =  Fact.all_ignore [
     Derive.segments;
     Derive.sections;
     Derive.segments;
   ]    
```